application/common/components/personnel/PersonnelUser.php

```diff
@@ -41,4 +41,9 @@ class PersonnelUser
      * @var null|string
      */
     public $spouseEmail;
-}
\ No newline at end of file
+
+    /**
+     * @var int
```

Comment:
I did make it `bool` at first, but came across this code in `updateProfileIfNeeded()`:
```
        foreach ($properties as $property => $value) {
            if ($this->$property != $value) {
                $dirty = true;
                $this->$property = $value;
            }
        }
```
I'd have to add a conditional with special handling of doNotDisclose.  It's a tradeoff, for sure.

Reply:
I believe it is because this foreach would assign `true` or `false` to the `User` model property `do_not_disclose`. When `save` is called, it would fail the type validation. I wonder... is there a way to override the validation and/or do the cast in the model?